Add pixi environments for Cavatica (and general development) (#863)
AlexsLemonade/scpca-nf: CONTRIBUTING.md

The `pixi.toml` file contains a list the high-level dependencies needed to run the workflow, and the `pixi.lock` file contains the exact versions of those dependencies that were used when the lockfile was created.

The included default environment includes `nextflow`, `nf-core`, `pre-commit`, and their dependencies.

To use this, you will need to have installed the `pixi` package manager, which is most easily done with the following command ([other installation methods](https://pixi.sh/latest/advanced/installation/)):

```bash
curl -fsSL https://pixi.sh/install.sh | bash
```

To use the default environment, you can then either prefix your command with `pixi run` (e.g. `pixi run nextflow`), or you can launch a shell with the environment activated by running `pixi shell`.
Either method will automatically update the environment to match the package versions defined the `pixi.toml` and/or `pixi.lock` files.

## Pre-commit hooks

For convenience, we have included a set of [pre-commit hooks](https://pre-commit.com/) that can be used to automatically format code according to the above specifications, as well as to spellcheck and check for other common errors.

> To use these hooks if you are using `pixi`, you can run `pixi run pre-commit install` from the `scpca-nf` directory.

Reviewer: ```suggestion
Once you have [installed `pixi`](#environment-management-with-pixi), you can run `pixi run pre-commit install` from the `scpca-nf` directory to use these hooks. 
```
Just some rewording and adding the link to installing pixi.
